zipline/data/data_portal.py

```diff
@@ -1017,7 +1050,8 @@ def _get_daily_window_for_sid(self, asset, field, days_in_window,
         else:
             return_array = np.zeros((bar_count,))
 
-        return_array[:] = np.NAN
+        if field != "volume":
+            return_array[:] = np.NAN
```

Comment:
We do nothing with volume because the array is already zeros, correct? Maybe a comment to that effect?